Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/algorithms.cpp

```diff
@@ -0,0 +1,45 @@
+#include "algorithms.hpp"
+
+namespace sne::saintNecPhysics
+{
+    std::vector<std::pair<PhysicObject *, PhysicObject *>> sweepAndPrune(std::vector<PhysicObject *> &v)
+    {
+        std::vector<std::pair<PhysicObject *, PhysicObject *>> pairs;
+
+        if (v.size() == 0)
+            return pairs;
+
+        // Sort object
+        std::sort(v.begin(), v.end(),
+                  [](PhysicObject *p1, PhysicObject *p2)
+                  {
+                      return p1->getPosition()[0] < p2->getPosition()[0]; // to update for min point axis
+                  });
+
+        glm::vec3 axis{1, 0, 0};
+        std::vector<PhysicObject *> curr;
+
+        for (int i = 0; i < v.size(); i++)
+        {
+            // For each object
+            for (std::vector<PhysicObject *>::iterator elt = curr.begin(); elt != curr.end(); elt++)
```

Comment:
I don't know if it is voluntary but here it copies an entire array each time

Reply:
Here we use iterator (pointer) so, if I am not wrong there is no copy